### PR TITLE
New Refresh API

### PR DESCRIPTION
**Pull request by SanjuPal01**
Signed-off-by: SanjuPal01 <sanju.sanju@progress.com>

### :nut_and_bolt: Description: What code changed, and why?
Created New Refresh API. When you hit this API with valid refresh token you will get new Refresh Token and ID token with new Expiry Time.

### :athletic_shoe: How to Build and Test the Change

`rebuild components/automate-dex`
`rebuild components/session-service`

For getting the refresh Token:
when you tried to login from automate ui. you can see the network tab, there is refresh api call from where you can get the `refresh-token` which you can use while using this new API.
```
{
    "refresh_token": "ChlyN2g1cXdtanY0ZmRmcT.....2bHFrbmkyaHdvcXh0dfNWI0M2pxbnFy",
    "grant_type": "refresh_token"
}
```
pass the refresh-token and grant-type as Body Parameter.

While making the call to this api also make sure to pass Authorization token. 
You can get the Authorization token from the request header of refresh API.

EndPoint: `https://automate-url/session/refresh_api`

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/automate-dex/habitat/plan.sh [hunk 1]
```diff
@@ -39,7 +39,7 @@ do_before() {
 do_unpack() {
   git clone "https://github.com/chef/dex-1" "$GOPATH/src/github.com/chef/dex-1"
   ( cd "$GOPATH/src/github.com/chef/dex-1" || exit
-    git checkout add_invalid_attempts
+    git checkout sanju/test_refresh
```
Comment (iamazzeez) [inline components/automate-dex/habitat/plan.sh:42]: raise a pr from dex-1 against add_invalid_attempts from sanju/test_refresh mention that pr in conversation

Comment (SanjuPal01) [reply components/automate-dex/habitat/plan.sh:42]: okay

File: components/session-service/oidc/oidc.go [hunk 2]
```diff
@@ -94,6 +98,14 @@ func New(cfg Config, retrySec int, certs *certs.ServiceCerts, l OIDCLogger) (Cli
...
 }
 
+func (c *client) RefreshTokenValidator(refreshToken string) (*http.Response, error) {
+	temp := myurl
+	temp += "/tokenValid"
+	formData := url.Values{}
+	formData.Add("refresh_token", refreshToken)
+	return c.Client().PostForm(temp, formData)
+}
+
```
Comment (iamazzeez) [inline components/session-service/oidc/oidc.go:108]: @SanjuPal01 variable name myurl could be changed to clientURL and that can be a part of client struct and use it in this function as c.clientURL, also code could be refactored with readable variable names, avoid using temp.

Comment (SanjuPal01) [reply components/session-service/oidc/oidc.go:108]: okay will do

**Comment by SanjuPal01** [comment]
Dex-1 PR: https://github.com/chef/dex-1/pull/81

File: components/automate-dex/habitat/plan.sh (hunk 1 above)
Comment (SanjuPal01) [reply components/automate-dex/habitat/plan.sh:42]: https://github.com/chef/dex-1/pull/81

**Comment by Dmaddu** [comment]
created another PR from the latest main https://github.com/chef/automate/pull/7280

**Pull request closed without merging**